### ISSUE TITLE
Incorrect type narrowing for isinstance when filter is a generic class with a type parameter that has a default

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7860]
```python
from typing import Generic
from typing_extensions import TypeVar


T = TypeVar("T", bound=int, default=int)


class Foo(Generic[T]): ...


def func(obj: object):
    if isinstance(obj, Foo):
        reveal_type(obj) # Should be Foo[Unknown], but is currently Foo[int]
```

**Comment by erictraut** [comment]
My first attempt at fixing this resulted in some [apparent regressions](https://github.com/microsoft/pyright/pull/7858#issuecomment-2097464160) that need to be investigated and addressed.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
